src/main/scala/fr/acinq/bitcoin/Script.scala

```diff
@@ -294,12 +299,16 @@ object Script {
 
   def isPayToScript(script: Array[Byte]): Boolean = script.length == 23 && script(0) == elt2code(OP_HASH160).toByte && script(1) == 0x14 && script(22) == elt2code(OP_EQUAL).toByte
 
-  def removeSignature(script: List[ScriptElt], signature: BinaryData): List[ScriptElt] = {
+  def removeSignature(script: List[ScriptElt], signature: BinaryData): (List[ScriptElt], Boolean) = {
     val toRemove = OP_PUSHDATA(signature)
-    script.filterNot(_ == toRemove)
+    (script.filterNot(_ == toRemove), script.contains(toRemove))
   }
 
-  def removeSignatures(script: List[ScriptElt], sigs: List[BinaryData]): List[ScriptElt] = sigs.foldLeft(script)(removeSignature)
+  def removeSignatures(script: List[ScriptElt], sigs: List[BinaryData]): (List[ScriptElt], Int) = sigs.foldLeft((script, 0)) {
+    case (acc, sig) =>
+      val (updated, found) = removeSignature(acc._1, sig)
+      (updated, if (found) acc._2 + 1 else acc._2)
+  }
 
```

Comment:
Same comment as above, plus if feels strange to use a counter here and a boolean in removeSignature (why not also a counter there ?)